cypress/e2e/createappointment.cy.ts

```diff
@@ -15,31 +15,46 @@ describe('Create New Appointment', () => {
         cy.contains('Appointments').should('exist');
     });
     it('create new appointment', () => {
-        cy.contains('New Appointment').should('exist').click();
-        cy.get('#demo-simple-select').click().get('li.MuiButtonBase-root:nth-child(10)').click();
-        cy.get('input#title').should('exist').type('help with laundry');
-        cy.get('input#location').should('exist').type('1123 star ave, Richmond');
-        cy.get('button.MuiButtonBase-root:nth-child(1) > svg:nth-child(1)')
-            .click()
-            .get('div.MuiDayPicker-weekContainer:nth-child(2) > button:nth-child(6)')
+        cy.contains('New Appointment').should('exist').click({ multiple: true });
+        cy.get('#demo-simple-select')
             .click()
-            .get('div.MuiPaper-root')
-            .click()
-            .get('.MuiClock-squareMask')
-            .click();
-        // cy.wait(2000);
-        cy.contains('button', 'Schedule').should('exist').click();
+            .get('li.MuiButtonBase-root:nth-child(10)')
+            .click({ multiple: true });
+        const timeNow = new Date().getTime();
+        cy.get('input#title').should('exist').type('help with groceries ').type(timeNow.toString());
+        cy.get('input#location').should('exist').type('1123 star ave, Richmond');
+        cy.get('button.MuiButtonBase-root:nth-child(1) > svg:nth-child(1)');
+        cy.get('input[placeholder="mm/dd/yyyy hh:mm (a|p)m"]')
+            .should('exist')
+            .clear()
+            .type('11/29/2022 03:00 PM');
+
+        cy.contains('button', 'Schedule').should('exist').click({ multiple: true });
         cy.contains('Appointments').should('exist');
-        cy.wait(2000);
+
         // Check if the appointment details are displayed on the list
-        cy.contains('help with laundry').should('exist');
+        cy.contains('help with gardening').should('exist');
         cy.contains('1123 star ave, Richmond').should('exist');
     });
     it('successfully view appointment details', () => {
-        cy.get('div.MuiBox-root:nth-child(2) > div:nth-child(1)').click();
+        cy.contains('View Appointment').first().should('exist').click();
         cy.wait(1000);
         cy.contains('Home').should('exist').click();
-        cy.wait(1000);
+    });
+    it('successfully delete appointment', () => {
+        cy.get('button').then(($a) => {
+            if ($a.children('svg')) {
+                const del = 'button[type="button"]>svg';
+                cy.get('[class*="css-t1nuxs"]')
+                    .first()
+                    .then(($x) => {
+                        const data1 = $x.text();
+                        cy.get(del).first().click({ multiple: true });
+                        // check if the appointment is deleted or not
+                        cy.get('[class*="css-t1nuxs"]').first().contains(data1).should('not.exist');
+                    });
+            }
```

Comment:
this is unnecessarily complicated and it is also prone to breakages. maybe we can do this another way? 

I looked at the html element. Seems like we can use the "DeleteIcon" test id. 

```
const APPOINTMENT_TITLE = 'help with gardening';

cy.contains(APPOINTMENT_TITLE).parent().find('[data-testid="DeleteIcon"]').click()
```